core-sok/src/core/filters/operator_filter.py

```diff
@@ -0,0 +1,82 @@
+from api.models.graph import Graph
+from .base_filter import Filter
+from api.models.node import Node
+
+
+class OperatorFilter(Filter):
+    """
+    OperatorFilter filters the graph so only nodes that contain a specific
+    attribute for which the given comparison returns true remain in the graph.
+    """
+
+    def __init__(self, attribute: str, operator: str, value: str):
+        """
+        Initializes the attribute filter.
+
+        :param attribute: The attribute to filter by.
+        :type attribute: str
+        :param operator: The operator to use.
+        :type operator: str
+        :param value: The value to compare to.
+        :type value: str
+        """
+        self.attribute = attribute
+        self.operator = operator
+        self.value = value
+    
+    def __eq__(self, __value: object) -> bool:
+        if not isinstance(__value, OperatorFilter):
+            return False
+        return self.attribute == __value.attribute and self.operator == __value.operator and self.value == __value.value
+
+    def satisfiesQuery(self, node: Node) -> bool:
+        """
+        Chcecks if the node satisfies the query.
+        :param node: The node to check.
+        :type node: Node
+        :return: True if the node satisfies the query, False otherwise. 
+        """
+        if self.attribute not in node.data:
+            return False
+
+        try:
+            self.value = type(node.data[self.attribute])(self.value)
+        except ValueError:
+            return False
+
+        if self.operator == "==":
+            return node.data[self.attribute] == self.value
+        if self.operator == "!=":
+            return node.data[self.attribute] != self.value
+        if self.operator == ">":
+            return node.data[self.attribute] > self.value
+        if self.operator == "<":
+            return node.data[self.attribute] < self.value
+        if self.operator == ">=":
+            return node.data[self.attribute] >= self.value
+        if self.operator == "<=":
+            return node.data[self.attribute] <= self.value
```

Comment:
The section of code handling different operators could be made in a more creative way, which can provide better possible expansion of the operators. This isn't really important, but if you want, check this link: 
https://stackoverflow.com/questions/1740726/turn-string-into-operator

I think you gonna like it :)

Reply:
Thank you, this will be a key resource on which I'll base the refactoring on.